crates/forge/bin/cmd/create.rs

```diff
@@ -157,126 +158,128 @@ impl CreateArgs {
+                                    panic!(
                                     "Could not find contract {} at path {} for compilation output",
                                     contract_name, contract_path)
-                                    });
-                                let fdep_fdeps_map = fdep_art
-                                    .factory_dependencies
-                                    .clone()
-                                    .expect("factory deps not found");
-                                for dep in fdep_fdeps_map.values() {
-                                    queue.push_back(dep.clone())
-                                }
-
-                                let fdep_bytecode = fdep_art
-                                    .bytecode
-                                    .clone()
-                                    .expect("Bytecode not found for factory dependency")
-                                    .object
-                                    .clone()
-                                    .into_bytes()
-                                    .unwrap()
-                                    .to_vec();
-                                visited_bytecodes.insert(fdep_bytecode);
+                                });
+                            let fdep_fdeps_map = fdep_art
+                                .factory_dependencies
+                                .clone()
+                                .expect("factory deps not found");
+                            for dep in fdep_fdeps_map.values() {
+                                queue.push_back(dep.clone())
                             }
+
+                            let fdep_bytecode = fdep_art
+                                .bytecode
+                                .clone()
+                                .expect("Bytecode not found for factory dependency")
+                                .object
+                                .clone()
+                                .into_bytes()
+                                .unwrap()
+                                .to_vec();
+                            visited_bytecodes.insert(fdep_bytecode);
                         }
-                        visited_bytecodes.insert(bytecode.clone());
-                        let factory_deps: Vec<Vec<u8>> = visited_bytecodes.into_iter().collect();
-                        let zk_data = ZkSyncData { bytecode, bytecode_hash, factory_deps };
-                        (abi, bin.object, Some(zk_data))
                     }
+
+                    visited_bytecodes.insert(bytecode.clone());
+                    let factory_deps: Vec<Vec<u8>> = visited_bytecodes.into_iter().collect();
+                    let zk_data = ZkSyncData { bytecode, bytecode_hash, factory_deps };
+                    Ok((abi, bin.object, Some(zk_data)))
                 };
 
+                let (abi, bytecode_object, zk_data) =
+                    output.as_mut().either(process_evm_output, process_zk_output)?;
```

Comment:
Can we use `either_into` here? (provided we don't need the output)

Reply:
we _do_ need the `output` in later iterations of the loop, and `either_into` calls `.into()` on L or R but we don't have that impl here